src/main/archetype/ui.frontend/webpack.prod.config.js

```diff
@@ -1,30 +1,32 @@
 'use strict';
 
-const path = require('path');
-const webpack = require('webpack');
-const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const path                    = require('path');
+const webpack                 = require('webpack');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const MiniCssExtractPlugin    = require("mini-css-extract-plugin");
 const OptimizeCSSAssetsPlugin = require("optimize-css-assets-webpack-plugin");
-const TSConfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
-const TSLintPlugin = require('tslint-webpack-plugin');
+const TSConfigPathsPlugin     = require('tsconfig-paths-webpack-plugin');
+const TSLintPlugin            = require('tslint-webpack-plugin');
 
-const FC_APPS_ROOT = __dirname + '/src/main/content/jcr_root/apps/${appsFolderName}';
-const CLIENTLIBS_PATH = FC_APPS_ROOT + '/clientlibs';
+const SOURCE_ROOT = __dirname + '/src/main/webpack';
+const FC_APPS_ROOT            = __dirname + '/../ui.apps/src/main/content/jcr_root/apps/${appsFolderName}';
+const CLIENTLIBS_PATH         = FC_APPS_ROOT + '/clientlibs';
 
 module.exports = (env) => {
     return {
         resolve: {
-            extensions: ['js', '.ts'],
+            extensions: ['js', '.ts', '.tsx'],
             plugins: [new TSConfigPathsPlugin({
                 configFile: "./tsconfig.json"
             })]
         },
         entry: {
-            site: CLIENTLIBS_PATH + '/clientlib-site/webpack/main.ts',
-            dependencies: CLIENTLIBS_PATH + '/clientlib-site/webpack/vendors.js'
+            site: SOURCE_ROOT + '/site/main.ts',
+            dependencies: SOURCE_ROOT + '/site/vendors.js'
         },
         output: {
             filename: (chunkData) => {
-                return chunkData.chunk.name === 'dependencies' ? 'clientlib-dependencies/[name].js': 'clientlib-site/[name].js';
+                return chunkData.chunk.name === 'dependencies' ? 'clientlib-dependencies/[name].js' : 'clientlib-site/[name].js';
             },
             path: CLIENTLIBS_PATH
```

Comment:
Another approach may be to use the [aem-clientlib-generator](https://www.npmjs.com/package/aem-clientlib-generator) to move the compiled CSS and Javascript into the ui.apps project. While it would add another config to the project, it is a little more robust and will simplify the webpack build...

Reply:
@godanny86 - @bpauli is PTO for the next few weeks, do you have any bandwidth to put together a code example for integrating this clientlib generator as you did with the spa archetype? I can also take over but I'm coming at this topic cold.

Reply:
Hi @richardhand, yes I can create some bandwidth this week. What is the easiest approach for sharing the code? I can fork the repo and perform a PR into the issue/124 branch?

Reply:
A PR against the `issue/124` would be great, thanks @godanny86!